src/runtime/set.cpp

```diff
@@ -114,7 +133,11 @@ Box* frozensetNew(Box* _cls, Box* container) {
 
     // Some optimizations from CPython: frozensets can be shared:
     if (_cls == frozenset_cls) {
-        if (!container) {
+        bool isEmpty = false;
+        if (container && PyObject_HasAttr(container, internStringMortal("__len__"))) {
```

Comment:
Minor, but it's nice to do something like `static BoxedString* len_str = internStringImmortal("__len__"); HasAttr(Container, len_str);` to avoid interning the string each time.

But also, I'm a bit worried about this optimization since CPython doesn't do it -- I think there's nothing stopping a user from defining a class where `__len__` returns 0 but iterating over it returns more-than-zero elements, and in that case we'll have different behavior than CPython.  Is there something where this helped a lot?